howtos/diffs/intermediate-layer-state.diff

```diff
@@ -0,0 +1,32 @@
+diff --git a/examples/mnist/train.py b/examples/mnist/train.py
+index 5dc74db..7affff9 100644
+--- a/examples/mnist/train.py
++++ b/examples/mnist/train.py
+@@ -65,7 +65,7 @@ class CNN(nn.Module):
+   """A simple CNN model."""
+ 
+   def apply(self, x):
+-    x = nn.Conv(x, features=32, kernel_size=(3, 3))
```

Comment:
Can we get an API that's like this:
```suggestion
    x = capture_shapes(nn.Conv)(x, features=32, kernel_size=(3, 3))
```

or like this:
```suggestion
    x = capture_shapes(nn.Conv(x, features=32, kernel_size=(3, 3)))
```

(the second seems cleaner as it doesn't require people to think about functions that take modules and return modules; but maybe you lose name information or something else that's helpful)